python/ray/experimental/data/impl/block_list.py

```diff
@@ -0,0 +1,33 @@
+from typing import Iterable, List, Optional, Any, Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pyarrow
+
+from ray.experimental.data.impl.block import Block, ObjectRef, T
+
+
+class BlockMetadata:
```

Comment:
Isn't `BlockMetadata` more of a per-block thing, i.e. shouldn't this be in `impl/block.py`? Especially since it'll be useful outside of just `BlockList` (I think).